main.go

```diff
@@ -58,6 +58,9 @@ func main() {
 	a.PUT("/rooms/:room/memberships", handlers.RoomMembershipsCreate)
 	a.DELETE("/rooms/:room/memberships", handlers.RoomMembershipsDelete)
 
+    a.PUT("/messages/:message/hearts", handlers.MessagesHeart)
```

Comment:
@pletcher Sorry for leaving this so long.

I'm not sure. It'd be an unnecessary overhead to actually check that the room exists and that the message is in the specified room, but if we don't, multiple URLs can perform the same action. So

    /room/BogusRoom/messages/actualMessageId/heart

Would work.

I'm wondering to what extent that might be a problem. The main disadvantage I currently see is that it would make request logs slightly harder to search through, because you'd have to take into account people may enter anyhing as a room slug.